### PR TITLE
Add tests for deprecated MDx code

### PR DESCRIPTION
**Pull request by k-stachowiak**
## Description
This PR enables execution of deprecated functions form the MDx API.

## Status
READY

## Requires Backporting
May be backported

## Migrations
NO

## Additional comments
None

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

## Steps to test or reproduce
None

### CODE REVIEW

**Comment by k-stachowiak** [comment]
@andresag01 The purpose of this PR is to change it in the MDx modules. I have incorrectly named and described it as concerning the entire hashing API. It is intentionally limited in order to ease continuous merging of the work step-by-step.

I have addressed your comment regarding the parentheses around the `sizeof`'s argument. Also I made sure more deprecated functions are called as in the previous versions most of them were actually cut off by the way `mbedtls_mdx( )` are implemented.

File: tests/suites/test_suite_mdx.function [hunk 1]
```diff
@@ -24,6 +28,17 @@ void md2_text( char *text_src_string, char *hex_hash_string )
     hexify( hash_str, output, sizeof  output );
 
     TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+    memset( hash_str, 0x00, sizeof(hash_str) );
```
Comment (andresag01) [inline tests/suites/test_suite_mdx.function]: Style: Please use spaces before and after `)` and `(`

File: tests/suites/test_suite_shax.function [hunk 2]
```diff
@@ -22,6 +26,17 @@ void mbedtls_sha1( char *hex_src_string, char *hex_hash_string )
     hexify( hash_str, output, 20 );
 
     TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+    memset(hash_str, 0x00, 10000);
```
Comment (andresag01) [inline tests/suites/test_suite_shax.function]: Style

File: tests/suites/test_suite_shax.function [hunk 3]
```diff
@@ -22,6 +26,17 @@ void mbedtls_sha1( char *hex_src_string, char *hex_hash_string )
...
+    memset(output, 0x00, 41);
```
Comment (andresag01) [inline tests/suites/test_suite_shax.function]: Style

Comment (andresag01) [reply tests/suites/test_suite_shax.function]: You could also use sizeof

File: tests/suites/test_suite_shax.function (hunk 2 above)
Comment (andresag01) [reply tests/suites/test_suite_shax.function]: You could also use sizeof

File: tests/suites/test_suite_shax.function [hunk 4]
```diff
@@ -106,6 +170,17 @@ void mbedtls_sha512(char *hex_src_string, char *hex_hash_string )
     hexify( hash_str, output, 64 );
 
     TEST_ASSERT( strcmp( (char *) hash_str, hex_hash_string ) == 0 );
+
+#if !defined(MBEDTLS_DEPRECATED_REMOVED)
+    memset(hash_str, 0x00, 10000);
+    memset(output, 0x00, 129);
+    mbedtls_sha512_init( &ctx );
+    mbedtls_sha512_starts( &ctx, 0);
```
Comment (andresag01) [inline tests/suites/test_suite_shax.function]: Style

**Comment by k-stachowiak** [comment]
@andresag01 It was a mistake on my side not to document the reasoning behind that commit, even though it is not obvious. From the functional standpoint it would have a non-zero value to leave the calls to deprecated API in, however:
* e.g. the SHA224 test calls a SHA256 function with `is224` set to 1, but this doesn't impact execution of the deprecated functions, but rather impacts the flow of a function deeper in the call stack. But that particular path is already triggered by the earlier call to a non-deprecated `mbedtls_sha224_ret( )`. Therefore, leaving the test code in, we would not trigger any new code path.
* It could still make sense to test the calls to make sure, that they actually set the is224 variable to 1. However, it would mean, we introduce ~10 lines of test code to test 1. Secondly the API is deprecated, and therefore the test will become obsolete in the future anyway.

**Comment by k-stachowiak** [comment]
@andresag01, @sbutcher-arm  I have resolved conflicts, which, unfortunately, required doing a rebase. Could you please re-review the PR?

**Comment by tom-cosgrove-arm** [comment]
Closing this as the void functions were removed in #4334

**Pull request closed without merging**